torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2356,14 +2355,21 @@ c10::intrusive_ptr<Work> ProcessGroupNCCL::endCoalescing() {
   c10::cuda::CaptureStatus capture_status =
       c10::cuda::currentStreamCaptureStatusMayInitCtx();
 
+  // Notify graphs before we check the capture status preemptively
+  at::cuda::CUDAGraph::inc_pending_event_queries();
```

Comment:
where can i learn more about the cudagraph pending event stuff?  cc @eellison

Reply:
Here: https://github.com/pytorch/pytorch/pull/111875
It is a working alternative to a neater solution: https://github.com/pytorch/pytorch/pull/105182
#105182 did not pass CI because CI are on old GPU arch (M60), and certain CUDA Graph APIs failed on them. (But no issue with newer GPUs).
I think we should move to #105182 someday.